src/Compression/CompressionCodecZSTD.cpp

```diff
@@ -79,9 +93,31 @@ void registerCodecZSTD(CompressionCodecFactory & factory)
 
             level = literal->value.safeGet<UInt64>();
             if (level > ZSTD_maxCLevel())
-                throw Exception("ZSTD codec can't have level more that " + toString(ZSTD_maxCLevel()) + ", given " + toString(level), ErrorCodes::ILLEGAL_CODEC_PARAMETER);
+                throw Exception(
+                    "ZSTD codec can't have level more that " + toString(ZSTD_maxCLevel()) + ", given " + toString(level),
+                    ErrorCodes::ILLEGAL_CODEC_PARAMETER);
+            if (arguments->children.size() > 1)
+            {
+                const auto * window_literal = children[1]->as<ASTLiteral>();
+                if (!window_literal)
```

Comment:
You're not checking if this is really an integer at this point

Reply:
hm, this code does the same as other compression codecs (and same what is done for first parameters in zstd codec, above)